diff -bBrU3 bitcoin-0.6.0-linux/src/src/main.cpp bitcoin-0.6.0-linux-patched/src/src/main.cpp
--- bitcoin-0.6.0-linux/src/src/main.cpp	2012-03-30 04:48:25.000000000 +0900
+++ bitcoin-0.6.0-linux-patched/src/src/main.cpp	2012-05-04 07:39:52.541380751 +0900
@@ -1648,6 +1648,16 @@
         if (!tx.CheckTransaction())
             return DoS(tx.nDoS, error("CheckBlock() : CheckTransaction failed"));
 
+    // Check for duplicate txids. This is caught by ConnectInputs(),
+    // but catching it earlier avoids a potential DoS attack:
+    set<uint256> uniqueTx;
+    BOOST_FOREACH(const CTransaction& tx, vtx)
+    {
+        uniqueTx.insert(tx.GetHash());
+    }
+    if (uniqueTx.size() != vtx.size())
+        return DoS(100, error("CheckBlock() : duplicate transaction"));
+
     int nSigOps = 0;
     BOOST_FOREACH(const CTransaction& tx, vtx)
     {
