diff -rbBU3 bitcoin-0.3.15/src/init.cpp bitcoin-0.3.15-modified/src/init.cpp
--- bitcoin-0.3.15/src/init.cpp	2010-11-13 10:01:00.000000000 +0900
+++ bitcoin-0.3.15-modified/src/init.cpp	2010-11-15 14:07:02.213191001 +0900
@@ -74,13 +74,16 @@
 #ifndef GUI
 int main(int argc, char* argv[])
 {
-    for (int i = 1; i < argc; i++)
+    fDaemon = false;
+    for (int i = 1; i < argc; i++) {
         if (!IsSwitchChar(argv[i][0]))
             fCommandLine = true;
-    fDaemon = !fCommandLine;
+        if (strcmp(argv[i], "-daemon") == 0)
+            fDaemon = true;
+    }
 
 #ifdef __WXGTK__
-    if (!fCommandLine)
+    if (fDaemon)
     {
         // Daemonize
         pid_t pid = fork();
diff -rbBU3 bitcoin-0.3.15/src/makefile.unix bitcoin-0.3.15-modified/src/makefile.unix
--- bitcoin-0.3.15/src/makefile.unix	2010-11-13 10:01:00.000000000 +0900
+++ bitcoin-0.3.15-modified/src/makefile.unix	2010-11-15 14:12:49.495723583 +0900
@@ -2,30 +2,22 @@
 # Distributed under the MIT/X11 software license, see the accompanying
 # file license.txt or http://www.opensource.org/licenses/mit-license.php.
 
+# Avoid using wx-config when not needed
+INCLUDEPATHS=-I"/usr/include/db4.7"
+ifeq ($(USE_WX),1)
+INCLUDEPATHS+=$(shell wx-config-2.9 --cflags)
+WXLIBS=$(shell wx-config-2.9 --libs)
+endif
 
-INCLUDEPATHS= \
- -I"/usr/local/include/wx-2.9" \
- -I"/usr/local/lib/wx/include/gtk2-unicode-debug-static-2.9"
-
-# for wxWidgets 2.9.1, add -l Xxf86vm
-WXLIBS= \
- -Wl,-Bstatic \
-   -l wx_gtk2ud-2.9 \
- -Wl,-Bdynamic \
-   -l gtk-x11-2.0 \
-   -l SM
-
-# for boost 1.37, add -mt to the boost libraries
 LIBS= \
- -Wl,-Bstatic \
+   -dead_strip \
    -l boost_system \
    -l boost_filesystem \
    -l boost_program_options \
    -l boost_thread \
-   -l db_cxx \
+   -l db_cxx-4.7 \
    -l ssl \
    -l crypto \
- -Wl,-Bdynamic \
    -l gthread-2.0 \
    -l z \
    -l dl
