diff -rbBU3 bitcoin-0.3.21-original/src/makefile.unix bitcoin-0.3.21/src/makefile.unix
--- bitcoin-0.3.21-original/src/makefile.unix	2011-05-18 16:03:05.000000000 +0900
+++ bitcoin-0.3.21/src/makefile.unix	2011-05-18 16:05:44.000000000 +0900
@@ -4,22 +4,23 @@
 
 CXX=g++
 
-WXINCLUDEPATHS=$(shell wx-config --cxxflags)
-
+WXINCLUDEPATHS=-I"/usr/include/db4.7"
+ifeq ($(USE_WX),1)
+WXINCLUDEPATHS+=$(shell wx-config-2.9 --cxxflags)
 WXLIBS=$(shell wx-config --libs)
+endif
 
-USE_UPNP:=0
 
 DEFS=-DNOPCH -DFOURWAYSSE2 -DUSE_SSL
 
 # for boost 1.37, add -mt to the boost libraries
 LIBS= \
- -Wl,-Bstatic \
+   -dead_strip \
    -l boost_system \
    -l boost_filesystem \
    -l boost_program_options \
    -l boost_thread \
-   -l db_cxx \
+   -l db_cxx-4.7 \
    -l ssl \
    -l crypto
 
@@ -29,7 +30,6 @@
 endif
 
 LIBS+= \
- -Wl,-Bdynamic \
    -l gthread-2.0 \
    -l z \
    -l dl
