diff -bBrU3 php-mongo-9999_orig/config.m4 php-mongo-9999/config.m4
--- php-mongo-9999_orig/config.m4	2009-07-30 21:43:55.000000000 +0900
+++ php-mongo-9999/config.m4	2009-07-30 21:44:01.000000000 +0900
@@ -10,9 +10,5 @@
 
   PHP_ADD_EXTENSION_DEP(mongo, date, false)
   PHP_ADD_EXTENSION_DEP(mongo, spl, false)
-
-  if test "$PHP_64" != "no"; then
-    CFLAGS="$CFLAGS -arch i386 -arch x86_64 -arch ppc -arch ppc64"
-  fi
 fi
 
